src/nvim/eval.c

```diff
@@ -15509,6 +15451,35 @@ static void f_sort(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   do_sort_uniq(argvars, rettv, true);
 }
 
+/// "stdioopen()" function
+static void f_stdioopen(typval_T *argvars, typval_T *rettv, FunPtr fptr)
```

Comment:
jobstart always starts a process. `sockconnect()` always connects to some socket, possibly over the network, at least some external process will be notified somewhere. `stdioopen()` has no immediately external effects, just sets the mode of operation of known fds.  `sockconnect()` and `stdioopen()` is as different from each other as to `jobstart()`. And yes in actual _behavior_, not merely "how unix names things".